### ISSUE TITLE
2.3.4 version adds escape for slashes

### ISSUE THREAD

**Issue by DavidePastore** [slimphp/Slim#663]
I'm working with Twig. In the base page I have something like this:

```
{%
 set links = {
  'Home': urlFor('HomePage'),
  'Link1': urlFor('Link1'),
  'Link2': urlFor('Link2'),
  'Link3': urlFor('Link3')
 }
%}
{% for key, link in links %}
 <li>{{ link }}</li>
{% endfor %}
```

This code works fine in 2.3.3, but in 2.3.4 it adds escape backslashes.
HomePage is defined in this way:

```
$app->get('/', function () use($app){
  $app->render('index.twig');
})->name("HomePage");
```

The other routes are similar to this.

What I get:

```
/\/
/\/link1
/\/link2
/\/link3
```

**Comment by erizocosmico** [comment]
I tried to reproduce your error but I don't get your results.

``` php
$app->get('/', function () use($app){
  $app->render('index.html');
})->name("HomePage");

$app->get('/hi/testing', function () use($app){
  $app->render('index.html');
})->name("Link1");
```

``` html
{%
 set links = {
  'Home': urlFor('HomePage'),
  'Link1': urlFor('Link1'),
  'Link2': urlFor('HomePage'),
  'Link3': urlFor('HomePage')
 }
%}
{% for key, link in links %}
 <li>{{ link }}</li>
{% endfor %}
```

And I get:

```
/
/hi/testing
/
/
```

No escape backslash is added for me.

Versions:

```
installed:
  slim/slim  2.3.4   Slim Framework, a PHP micro framework
  slim/views 0.1.0   Twig View Parser package for the Slim Framework
  twig/twig  v1.14.1 Twig, the flexible, fast, and secure template language f...
```

**Comment by DavidePastore** [comment]
I tried with your versions but the problem is always here. It's so strange.

**Comment by codeguy** [comment]
I'll touch base with @silentworks and see if the Slim-Views repo needs any tweaks for the 2.3.4 update.

**Comment by silentworks** [comment]
Hi @DavidePastore can you state your OS and PHP version please?

**Comment by DavidePastore** [comment]
Hi @silentworks.

OS: Microsoft Windows XP Home Edition Service Pack 3.
PHP: 5.3.8 on xampp.

**Comment by tuupola** [comment]
Hardcoded "/" and DIRECTORY_SEPARATOR clashing somewhere?

**Comment by DavidePastore** [comment]
I have not touched any of this.

**Comment by tuupola** [comment]
I ment in Slim or Twig code.

**Comment by DavidePastore** [comment]
Unfortunately, everything is pretty standard. I will try to create a project in which I show where the problem arises.

**Comment by codeguy** [comment]
@DavidePastore That'd be great. I'll use that on a local Windows VM to duplicate the issue you are seeing.

**Comment by DavidePastore** [comment]
@codeguy This is all. Basic example where I found the bug. https://github.com/DavidePastore/SlimFrameworkBackslashIssue

**Comment by silentworks** [comment]
Yeah I have tested this and getting the same issue, this even occurs with the Slim View, so its an issue within urlFor method. I will take a closer look into this.

**Comment by codeguy** [comment]
Should be fixed in 2.3.5. If not let me know.
